src/Microsoft.AspNetCore.Routing/InvalidRouteException.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public class InvalidRouteException : Exception
```

Comment:
I was thinking that if we might use this exception in some other cases we might need a more general name(assuming we might not prefer to create more custom exception types).